### PR TITLE
External networks mapping option

### PR DESCRIPTION
**Pull request by toha10**
Added external networks mapping option (ext_net_map in config file).
This option allow to map external networks on source cloud to
existing external networks on destination cloud with appropriate parameters by
dictionary with old and new ids.

### CODE REVIEW

**Comment by Wasapl** [comment]
Merged build finished. Test FAILed.

**Comment by Wasapl** [comment]
Merged build finished. Test PASSed.

File: cloudferrylib/os/network/neutron.py [hunk 1]
```diff
@@ -37,6 +37,8 @@ def __init__(self, config, cloud):
         self.cloud = cloud
         self.identity_client = cloud.resources['identity']
         self.neutron_client = self.proxy(self.get_client(), config)
+        self.ext_net_map = utl.read_yaml_file(self.config.migrate.ext_net_map) \
+            if utl.read_yaml_file(self.config.migrate.ext_net_map) else {}
```
Comment (roman-verchikov) [inline cloudferrylib/os/network/neutron.py]: `self.ext_net_map = utl.read_yaml_file(self.config.migrate.ext_net_map) or {}` would be much cleaner.

**Comment by roman-verchikov** [comment]
Would be great if you could isolate external networks logic into separate class and cover it with unit tests. At the very least let's have the introduced logic being tested at least.

File: cloudferrylib/os/network/neutron.py (hunk 1 above)
Comment (toha10) [inline cloudferrylib/os/network/neutron.py]: Thank you for the notice

**Comment by Wasapl** [comment]
Merged build finished. Test PASSed.

**Comment by toha10** [comment]
Speaking about external networks logic - it's just mapping that is used in case if we have already created external networks on dest cloud and we want to use these networks instead of migration the old ones.
Could you please clarify: are you proposing to isolate the whole external networks getting logic ?

**Comment by roman-verchikov** [comment]
> are you proposing to isolate the whole external networks getting logic ?

Yeah, but that's probably too much of an effort, so let's just cover introduced logic with unit tests for now.

**Comment by plomakin** [comment]
+1 doesn't affect common functionality

**Comment by roman-verchikov** [comment]
+1

**Comment by roman-verchikov** [comment]
Two +1s, can be merged

**Pull request merged**